### ISSUE TITLE
PHP Errors on page template

### ISSUE THREAD

**Issue by shmaltz** [soflyy/oxygen-bugs-and-features#1299]
**Describe the bug**
*A clear and concise description of what the bug is.*
I am getting these PHP errors on my Single Post Page (via Query Monitor Plugin):

```
Warning | strpos() expects parameter 1 to be string, array given | 2 | +wp-content/plugins/oxygen/component-framework/components/component.class.php:3602 | Plugin: oxygen
-- | -- | -- | -- | --
Warning | strpos() expects parameter 1 to be string, array given | 2 | +wp-content/plugins/oxygen/component-framework/components/component.class.php:3606 | Plugin: oxygen
Warning | strpos() expects parameter 1 to be string, array given | 2 | +wp-content/plugins/oxygen/component-framework/components/component.class.php:3610 | Plugin: oxygen
Warning | strpos() expects parameter 1 to be string, array given | 2 | +wp-content/plugins/oxygen/component-framework/components/component.class.php:3614 | Plugin: oxygen
Warning | strpos() expects parameter 1 to be string, array given | 2 | +wp-content/plugins/oxygen/component-framework/components/component.class.php:3618 | Plugin: oxygen
Warning | strpos() expects parameter 1 to be string, array given | 2 | +wp-content/plugins/oxygen/component-framework/components/component.class.php:3622 | Plugin: oxygen
Notice | Undefined index: preview | 1 | +wp-content/plugins/oxygen/component-framework/components/classes/comments-list.class.php:79 | Plugin: oxygen
Notice | Undefined index: preview | 1 | +wp-content/plugins/oxygen/component-framework/components/classes/comments-list.class.php:91 | Plugin: oxygen
Notice | Undefined index: preview | 1 | +wp-content/plugins/oxygen/component-framework/components/classes/comments-list.class.php:102 | Plugin: oxygen
Notice | Undefined index: action | 1 | +wp-content/plugins/oxygen/component-framework/components/classes/comments-list.class.php:102 | Plugin: oxygen
Notice | Undefined index: preview | 1 | +wp-content/plugins/oxygen/component-framework/components/classes/comment-form.class.php:75 | Plugin: oxygen
Notice | Undefined index: preview | 1 | +wp-content/plugins/oxygen/component-framework/components/classes/comment-form.class.php:79 | Plugin: oxygen
Notice | Undefined index: button-text-color | 1 | +wp-content/plugins/oxygen/component-framework/components/component.class.php:3035 | Plugin: oxygen
Notice | Undefined index: button-size | 1 | +wp-content/plugins/oxygen/component-framework/components/component.class.php:3036 | Plugin: oxygen
Notice | Array to string conversion | 1 | +wp-content/plugins/oxygen/component-framework/component-init.php:4079 | Plugin: oxygen
Deprecated | implode(): Passing glue string after array is deprecated. Swap the parameters | 112 | +wp-content/plugins/oxygen/component-framework/component-init.php:4846 | Plugin: oxygen
Deprecated | join(): Passing glue string after array is deprecated. Swap the parameters | 26 | +wp-content/plugins/oxygen/component-framework/components/component.class.php:2281 | Plugin: oxygen
```

**Have you tried all the steps at https://oxygenbuilder.com/documentation/troubleshooting/troubleshooting-guide/?**
Yes

**Are you able to replicate the issue on a Sandbox install at https://oxygenbuilder.com/try?**
*If yes, provide the link to a Sandbox install where the issue is present. If the issue only exists on a specific post or template, provide a direct link to that post or template.*
No
**To Reproduce**
I am using a local server so I cannot show you my side, but I created a new Single Post template, styled it and added dynamic content.

**Expected behavior**
No errors.

**What actually happens**
Errors.

**Screenshots**
Code pasted above

**Desktop (please complete the following information):**
- OS: Windows
 - Browser Chrome
 - Oxygen Version 3.3.7

**Comment by shmaltz** [comment]
@KittenCodes Thanks for marking this as a confirmed bug.

Do you of any work around to get rid of this error until it is fixed? Or what is causing it?

It is causing my pages to take ~3.5 seconds to load.

Thanks

**Comment by shoelaced** [comment]
Most of these are happening in 3.4 too.

**Comment by shmaltz** [comment]
Update: I updated to 3.4 thinking it might help, but didn't. It seems this error is being generated by the header builder. Once I deleted the header builder from my template, the PHP errors disappeared. A few PHP notices did remain though, but no big deal. We gotta fix this Header Builder :)

**Comment by shoelaced** [comment]
I'm actually not using the Header Builder - I thought it was the Pro Menu, which is why I was anticipating 3.4, as it mentions the strpos warnings in the changelog. But when I looked into it, it referenced some function that had to do with returning CSS values.

**Comment by shmaltz** [comment]
Very likely the pro menu, because I had that in the header builder.

**Comment by eizil** [comment]
Workaround for this (at least what works for me).

If you use pro menu, delete the pro menu, and add it back if you still want to use it.

Then the error will be fixed. Seems like when you first add the pro menu, some settings are missing.

**Comment by adatfalo** [comment]
Hello!

I have a very similar problem, but I don't use Pro Menu. I recreated the simple Menu, but it doesn't helped.

I get these warnings:

[09-Aug-2020 23:14:28 UTC] PHP Warning: strpos() expects parameter 1 to be string, array given in /home/tfaidg/public_html/sngy/wp-content/plugins/oxygen/component-framework/components/component.class.php on line 3622
[09-Aug-2020 23:14:28 UTC] PHP Warning: strpos() expects parameter 1 to be string, array given in /home/tfaidg/public_html/sngy/wp-content/plugins/oxygen/component-framework/components/component.class.php on line 3626
[09-Aug-2020 23:14:28 UTC] PHP Warning: strpos() expects parameter 1 to be string, array given in /home/tfaidg/public_html/sngy/wp-content/plugins/oxygen/component-framework/components/component.class.php on line 3630
[09-Aug-2020 23:14:28 UTC] PHP Warning: strpos() expects parameter 1 to be string, array given in /home/tfaidg/public_html/sngy/wp-content/plugins/oxygen/component-framework/components/component.class.php on line 3634
[09-Aug-2020 23:14:28 UTC] PHP Warning: strpos() expects parameter 1 to be string, array given in /home/tfaidg/public_html/sngy/wp-content/plugins/oxygen/component-framework/components/component.class.php on line 3638
[09-Aug-2020 23:14:28 UTC] PHP Warning: strpos() expects parameter 1 to be string, array given in /home/tfaidg/public_html/sngy/wp-content/plugins/oxygen/component-framework/components/component.class.php on line 3642

What else can cause it?

**Comment by irfan-muhammad** [comment]
Hi

I am using Oxygen Builder Version 3.1.1and  getting below warning:

`NOTICE: PHP message: PHP Warning:  strpos() expects parameter 1 to be string, array given in /wp-content/plugins/oxygen/component-framework/components/component.class.php on line 3564`

How i can fix it?

**Comment by techjase** [comment]
Hi I'm also getting these warnings. [Wed Aug 19 14:37:52 2020] [error] [client] AH01071: Got error 'PHP message: PHP Warning:  strpos() expects parameter 1 to be string, array given in /home/sites/9a/b/b630998959/public_html/wp-content/plugins/oxygen/component-framework/components/component.class.php on line 3622PHP message: PHP Warning:  strpos() expects parameter 1 to be string, array given in /home/sites/9a/b/b630998959/public_html/wp-content/plugins/oxygen/component-framework/components/component.class.php on line 3626PHP message: PHP Warning:  strpos() expects parameter 1 to be string, array given in /home/sites/9a/b/b630998959/public_html/wp-content/plugins/oxygen/component-framework/components/component.class.php on line 3630PHP message: PHP Warning:  strpos() expects parameter 1 to be string, array given in /home/sites/9a/b/b630998959/public_html/wp-content/plugins/oxygen/component-framework/components/component.class.php on line 3634

I'm using Oxygen 3.5 RC1, Wordpress 5.4.2 (waiting for 100% plugin compatibility). Will these warning cause me an issue with my hosting company?

**Comment by andergmartins** [comment]
Same here. Having tons of those messages on my logs. 
Duplicated issue: #1448

**Comment by rcg4u** [comment]
3.6 beta 1 fixes this error its in the change log https://oxygenbuilder.com/2020/09/28/oxygen-3-6-beta-1-now-available

**Comment by KittenCodes** [comment]
Fixed in v3.6.
